File: NHSOnline.Backend.Worker.Mocking/CID/Models/TokenRequest.cs
namespace NHSOnline.Backend.Worker.Mocking.Emis.Models
{
    public class TokenRequest
    {
        public string Grant_Type { get; }
        public string Code { get; }
        public string Redirect_Uri { get; }
        public string Code_Verifier { get; }
        public string Client_Id { get; }
        public string Code_Challenge_Method { get; }


        public TokenRequest(string code, string code_verifier)
        {
            Grant_Type = "authorization_code";
            Code = code;
            Redirect_Uri = "http://localhost:3000/auth-return";
            Code_Verifier = code_verifier;
            Client_Id = "nhs-online-poc";
            Code_Challenge_Method = "S256";
        }
    }
}

File: src/main/kotlin/mocking/citizenId/models/TokenRequest.kt
package mocking.citizenId.models

data class TokenRequest(
        val codeVerifier: String,
        val grantType: String = "authorization_code",
        val redirectUri: String = "http://localhost:3000/auth-return",
        val clientId: String = "nhs-online-poc",
        val codeChallengeMethod: String = "S256",
        val code: String?
)